controller/src/beerocks/master/db/db.cpp

```diff
@@ -2423,73 +2423,257 @@ const std::list<sChannelScanResults> &db::get_channel_scan_results(const sMacAdd
 //
 std::chrono::steady_clock::time_point db::get_client_persistent_data_last_edit(const sMacAddr &mac)
 {
-    return std::chrono::steady_clock::time_point::min();
+    auto n = get_node(mac);
+    if (!n) {
+        LOG(ERROR) << "node not found.... ";
+        return std::chrono::steady_clock::time_point::min();
+    } else if (n->get_type() != beerocks::TYPE_CLIENT) {
+        LOG(ERROR) << __FUNCTION__ << "node " << mac << " is not a valid client!";
+        return std::chrono::steady_clock::time_point::min();
+    }
+
+    return n->client_persistent_data_last_edit;
 }
 
 bool db::set_client_time_life_delay(const sMacAddr &mac,
                                     const std::chrono::seconds &time_life_delay_sec,
                                     bool set_persistent_db)
 {
+    auto n = get_node(mac);
+    if (!n) {
+        LOG(ERROR) << "node not found.... ";
+        return false;
+    } else if (n->get_type() != beerocks::TYPE_CLIENT) {
+        LOG(ERROR) << __FUNCTION__ << "node " << tlvf::mac_to_string(mac)
+                   << " is not a valid client!";
+        return false;
+    }
+
+    LOG(DEBUG) << __FUNCTION__ << ", time_life_delay_sec = " << time_life_delay_sec.count();
+
+    auto timestamp = std::chrono::steady_clock::now();
+    if (set_persistent_db) {
+        LOG(DEBUG) << __FUNCTION__ << ", configuring persistent-db, time_life_delay_sec = "
+                   << time_life_delay_sec.count();
+    }
+
+    n->client_time_life_delay_sec       = time_life_delay_sec;
+    n->client_persistent_data_last_edit = timestamp;
```

Comment:
AFAIU - the naming here is misleading `client_persistent_data_last_edit` - in the case that `set_persistent_db` is set to false you are not actually updating the persistent DB, you only update the controller runtime DB.

Reply:
I also don't understand. I thought that if `set_persistent_db` is set, we should save something on the persistent db.

Reply:
@simonbaren 
client persistent data can be set to the runtime db with multiple calls and then a call to `update_client_persistent_db()` will sync the runtime configuration to the persistent DB. the timestamp is handling the time of last edit (of configuration) and not the last time of actual synchronization to the persistent DB.
in any case - as part of the orderly shutdown of the controller - the runtime configuration should be saved to the persistent db - but again - with the timestamp indicating the last time of client configuration edit and not when it was synched to the persistent DB!
and naming suggestion is welcomed

@morantr - this is correct, but I will quote the draft PR description:
"Draft PR - missing the calls to BPL APIs will be added after #1460 is merged"
(at time of the draft PR creation BPL interface was not available yet)